homeassistant/components/switch/child_switch.py

```diff
@@ -0,0 +1,252 @@
+"""
+A generic switch that can be created by other componenets during setup
```

Comment:
Cheers, I've been thinking about this for ages, I am intending to make one for sensors as well once the process for a switch is locked down implementing a child_sensor components becomes trivial.